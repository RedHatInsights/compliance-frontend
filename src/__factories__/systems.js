import { Factory } from 'fishery';
import { faker } from '@faker-js/faker';
import { id, hostname } from './helpers';
import { testResultProfiles, buildNonCompliantProfiles } from './profiles';

const systemsFactory = Factory.define(({ params, transientParams }) => {
  const [osMajorVersion, osMinorVersion] = transientParams.osVersions
    ? faker.helpers.arrayElement(transientParams.osVersions).split('.')
    : [];

  return {
    ...id(),
    name: hostname(),
    ...(osMajorVersion
      ? {
          osMajorVersion,
          osMinorVersion,
        }
      : {}),
    ...params,
  };
});

export const buildNonCompliantSystems = (count = 5) =>
  systemsFactory.buildList(count, {
    compliant: false,
    testResultProfiles: [
      ...buildNonCompliantProfiles(),
      ...testResultProfiles.buildList(5),
    ],
  });

export const buildUnsupportedSystems = (count = 5) =>
  systemsFactory.buildList(count, {
    compliant: false,
    testResultProfiles: [...buildNonCompliantProfiles(2, { supported: false })],
  });

export const buildSystemsWithTestResultProfiles = (count = 5, params = {}) =>
  systemsFactory.buildList(count, {
    compliant: params.compliant || false,
    testResultProfiles: [
      ...(!params.compliant ? buildNonCompliantProfiles() : []),
      ...testResultProfiles.buildList(5),
    ],
    ...params,
  });

export default systemsFactory;
