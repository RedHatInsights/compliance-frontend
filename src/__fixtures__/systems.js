export const systems = [
  {
    id: 'd5bc2459-21ce-4d11-bc0b-03ea7513dfa6',
    display_name: 'demo.example.com',
    end_time: '2019-11-21T14:32:19Z',
    failed_rule_count: 28,
    supported: true,
    score: 40.0,
    compliant: false,
    security_guide_version: '0.14.3',
    os_major_version: '7',
    os_minor_version: '8',
    system_id: 'd5bc2459-21ce-4d11-bc0b-03ea7513dfa6',
    type: 'test_result',
    groups: [
      {
        id: 'dc925b53-0bee-4ccf-b95a-4b9611362cf7',
        name: 'Ungrouped Hosts',
        ungrouped: true,
      },
    ],
    tags: [],
  },
  {
    id: 'c1f0aef3-66fd-4abc-aa33-ec6755dd39d9',
    display_name: 'foo.example.com',
    end_time: null,
    failed_rule_count: 0,
    supported: true,
    score: 0.0,
    compliant: false,
    security_guide_version: '0.14.5',
    os_major_version: '7',
    os_minor_version: '8',
    system_id: 'c1f0aef3-66fd-4abc-aa33-ec6755dd39d9',
    type: 'test_result',
    groups: [
      {
        id: 'dc925b53-0bee-4ccf-b95a-4b9611362cf7',
        name: 'Ungrouped Hosts',
        ungrouped: true,
      },
    ],
    tags: [],
  },
];
