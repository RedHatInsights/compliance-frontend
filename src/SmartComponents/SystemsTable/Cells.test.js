import {
  Name,
  SSGVersion,
  Policies,
  FailedRules,
  ComplianceScore,
  LastScanned,
} from './Cells';
import { systems } from '@/__fixtures__/systems.js';
const testSystem = systems[0].node;

describe('Name', () => {
  it('returns', () => {
    expect(renderJson(<Name {...testSystem} />)).toMatchSnapshot();
  });
});

describe('SSGVersion', () => {
  it('returns', () => {
    expect(renderJson(<SSGVersion {...testSystem} />)).toMatchSnapshot();
  });

  it('returns no error without testResultProfiles', () => {
    expect(
      renderJson(<SSGVersion {...testSystem} testResultProfiles={[]} />)
    ).toMatchSnapshot();
  });
});

describe('Policies', () => {
  it('returns', () => {
    expect(renderJson(<Policies {...testSystem} />)).toMatchSnapshot();
  });

  it('returns without error without policies', () => {
    expect(
      renderJson(<Policies {...testSystem} policies={undefined} />)
    ).toMatchSnapshot();
  });
});

describe('FailedRules', () => {
  it('returns', () => {
    expect(renderJson(<FailedRules {...testSystem} />)).toMatchSnapshot();
  });

  it('returns no error without testResultProfiles', () => {
    expect(
      renderJson(<FailedRules {...testSystem} testResultProfiles={[]} />)
    ).toMatchSnapshot();
  });
});

describe('ComplianceScore', () => {
  it('returns', () => {
    expect(renderJson(<ComplianceScore {...testSystem} />)).toMatchSnapshot();
  });

  it('returns no error without testResultProfiles', () => {
    expect(
      renderJson(<ComplianceScore {...testSystem} testResultProfiles={[]} />)
    ).toMatchSnapshot();
  });
});

describe('LastScanned', () => {
  it('returns', () => {
    expect(renderJson(<LastScanned {...testSystem} />)).toMatchSnapshot();
  });

  it('returns NEVER', () => {
    expect(
      renderJson(<LastScanned {...testSystem} testResultProfiles={undefined} />)
    ).toMatchSnapshot();
  });

  it('returns no error without testResultProfiles', () => {
    expect(
      renderJson(<LastScanned {...testSystem} testResultProfiles={[]} />)
    ).toMatchSnapshot();
  });
});
