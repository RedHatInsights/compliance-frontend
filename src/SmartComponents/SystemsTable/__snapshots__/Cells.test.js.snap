// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComplianceScore returns 1`] = `
<ComplianceScore
  compliant={false}
  score={40}
  supported={true}
/>
`;

exports[`ComplianceScore returns no error without testResultProfiles 1`] = `"N/A"`;

exports[`FailedRules returns 1`] = `
<SystemLink
  id="d5bc2459-21ce-4d11-bc0b-03ea7513dfa6"
>
  28
</SystemLink>
`;

exports[`FailedRules returns no error without testResultProfiles 1`] = `
<SystemLink
  id="d5bc2459-21ce-4d11-bc0b-03ea7513dfa6"
>
  N/A
</SystemLink>
`;

exports[`LastScanned returns 1`] = `
<DateFormat
  date={1574346739000}
  type="relative"
/>
`;

exports[`LastScanned returns NEVER 1`] = `<NeverScanned />`;

exports[`LastScanned returns no error without testResultProfiles 1`] = `<NeverScanned />`;

exports[`Name returns 1`] = `
<TextContent>
  demo.example.com
</TextContent>
`;

exports[`Policies returns 1`] = `
<Truncate
  inline={true}
  length={215}
  text="HIPAA Policy"
/>
`;

exports[`Policies returns without error without policies 1`] = `""`;

exports[`SSGVersion returns 1`] = `
<UnsupportedSSGVersion
  messageVariant="singular"
>
  Not available
</UnsupportedSSGVersion>
`;

exports[`SSGVersion returns no error without testResultProfiles 1`] = `
<UnsupportedSSGVersion
  messageVariant="singular"
>
  Not available
</UnsupportedSSGVersion>
`;
